File: lib/memo_node.rb
class MemoNode
  attr_reader :title
  attr_reader :text
  attr_reader :parent
  attr_reader :children
  attr_reader :attr
  attr_reader :level

  def initialize(title, text, parent, attr, level)
    @title = title
    @text = text
    @parent = parent
    @children = []
    @attr = attr
    @level = level
  end

  def get_title()
    "[#{@attr}] #{@title}"
  end

  def append_node(node)
    @children.push(node)
  end

  def append_text(text)
    @text << "\n" << text
  end
end

File: lib/model/memo_node.go
package model

// MemoNode is
type MemoNode struct {
	Title     string
	Text      string
	Parent    *MemoNode
	Children  []*MemoNode
	Attribute string
	Level     int
}

// NewMemoNode is
func NewMemoNode(title string, text string, parent *MemoNode, children []*MemoNode, attribute string, level int) *MemoNode {
	return &MemoNode{
		Title:     title,
		Text:      text,
		Parent:    parent,
		Children:  children,
		Attribute: attribute,
		Level:     level,
	}
}

// AddChild is
func addChild(node *MemoNode, child *MemoNode) {
	node.Children = append(node.Children, child)
}

// AddText is
func addText(node *MemoNode, text string) {
	node.Text = node.Text + "\n" + text
}
